packages/vscode/src/plugins/prisma-postgres-manager/PrismaPostgresRepository.ts

```diff
@@ -159,22 +160,24 @@ export class PrismaPostgresApiRepository implements PrismaPostgresRepository {
 
     if (response.response.status === 401) {
       void this.removeWorkspace({ workspaceId })
-      throw new Error(`Credentials for workspace expired. Please login again.`)
+      throw new Error(`Session expired. Please sign in to continue.`)
     }
 
-    const error = response.error
-    if (typeof error === 'object' && error !== null) {
-      if ('error' in error && typeof error === 'object' && error !== null) {
-        if ('message' in error && typeof error.message === 'string') {
-          throw new Error(`API Error: ${error.message}`)
-        }
-      }
-      if ('errorDescription' in error && typeof error.errorDescription === 'string') {
-        throw new Error(`API Error: ${error.errorDescription}`)
-      }
-    }
+    const error = z
+      .object({
+        message: z.string().optional(),
+        errorDescription: z.string().optional(),
+        error: z
+          .object({
+            message: z.string().optional(),
+          })
+          .optional(),
+      })
+      .safeParse(response.error)
```

Comment:
ℹ️ This is a bit of "try to catch all" approach to error parsing. Some API errors return a object of shape `{ error: { code: "...", message: "..." } }`, some `{ error: "unauthorized", errorDescription: "..." }` and it might potentially be a regular `Error` class instance as well with a message property.